### PR TITLE
New changeling ability: Bone Shard

### PR DESCRIPTION
**Pull request by ItsMarmite**
## What Does This PR Do
This PR adds a new offensive changeling ability called Bone shard, heavily inspired by https://github.com/ParadiseSS13/Paradise/pull/14589.

Costing 15 chemicals per use, Bone shard allows changelings to create bone shards they can throw at their enemies, doing 20 damage per hit and embedding.

Once these bone shards have hit a foe, they will dull and be unusable. They however CAN be caught, and an aware target may be able to turn the bone shards against the attacking changeling if they're cautious.

## Why It's Good For The Game
Changelings have lacked any range for a while, outside of tentacle, which ultimately is less of a ranged weapon and more of a crowd control tool. I'm hoping this ability can lessen that gap between changelings and security whilst lessening reliance on armblade.

## Images of changes
image

https://github.com/ParadiseSS13/Paradise/assets/85680653/9ea347c3-ae54-40ee-8b6a-73544fe6070d

## Testing
see above. skrell rights activists are bashing down my door, I have embedded bone shards into so many. They scream for justice.

## Changelog
:cl:
add: Adds new changeling offensive ability, Bone shard- a series of bone shuriken to throw at your enemies.
/:cl:

### CODE REVIEW

**Comment by tipoima** [comment]
>Be cling
>Get random searched
>"Mr. Graytide, why do you have a full bag of bone shards"

**Comment by McRamon** [comment]
yeah i hope they decay and disspear rather quickly after being produced

**Comment by PopeDaveThe3th** [comment]
Maybe let clings eat them for quick evidence disposal.

**Comment by ItsMarmite** [comment]
> Maybe let clings eat them for quick evidence disposal.

I didn't demonstrate it in the video, but you can take them in hand and drop them to put them back inside you.

**Comment by tipoima** [comment]
I am thinking less about evidence and more about how balanced it would be to stock up on these without chemical cost in combat

**Comment by matttheficus** [comment]
Fine with these as long as you can’t stock a full box of them for later use.

**Comment by ItsMarmite** [comment]
> Fine with these as long as you can’t stock a full box of them for later use.

I'll increase their size on next commit to prevent this.

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 1]
```diff
@@ -582,3 +582,47 @@
...
+/obj/item/throwing_star/boneshard
+	name = "bone shard"
+	desc = "A serrated shard of bone laden with vicious barbs."
+	icon_state = "bone_star"
+	throwforce = 15
+	embedded_fall_chance = 5
+	embedded_impact_pain_multiplier = 3
+	embedded_unsafe_removal_pain_multiplier = 6
+	embedded_pain_chance = 10
+	materials = list(MAT_METAL=0, MAT_GLASS=0)
```
Comment (Henri215) [inline code/modules/antagonists/changeling/powers/mutations.dm:611]:
```suggestion
	materials = list(MAT_METAL = 0, MAT_GLASS = 0)
```

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 2]
```diff
@@ -582,3 +582,48 @@
 	armor = list(MELEE = 40, BULLET = 40, LASER = 40, ENERGY = 20, BOMB = 10, RAD = 0, FIRE = 90, ACID = 90)
 	flags_inv = HIDEEARS
 	flags_cover = MASKCOVERSEYES | MASKCOVERSMOUTH
+
+// Bone Shard
+
```
Comment (Burzah) [inline code/modules/antagonists/changeling/powers/mutations.dm:587]:
```suggestion
```

**Review by Burzah — APPROVED** [review]

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 3]
```diff
@@ -583,3 +583,47 @@
...
+	helptext = "The shards of bone will dull upon hitting a target, rendering them unusable as weapons."
+	button_icon_state = "boneshard"
+	chemical_cost = 15
+	dna_cost = 3
+	req_human = TRUE
+	weapon_type = /obj/item/throwing_star/boneshard
+	weapon_name_simple = "bone"
+	power_type = CHANGELING_PURCHASABLE_POWER
+	category = /datum/changeling_power_category/offence
+
```
Comment (DGamerL) [inline code/modules/antagonists/changeling/powers/mutations.dm:600]:
```suggestion
```

File: code/modules/antagonists/changeling/powers/mutations.dm [hunk 4]
```diff
@@ -583,3 +583,47 @@
...
+		embed_chance = 0
+
+/obj/item/throwing_star/boneshard/Initialize(mapload)
+	. = ..()
+	if(iscarbon(loc))
+		var/mob/living/carbon/C = loc
+		C.throw_mode_on()
+		if(loc)
+			playsound(loc, 'sound/effects/bone_break_1.ogg', 100, TRUE)
+			C.visible_message("<span class='warning'>Shards of bone grow through [C.name]'s palms and fall into [C.p_their()] hands!</span>", "<span class='warning'>We expel shards of bone into our hands.</span>", "<span class='hear'>You hear organic matter ripping and tearing!</span>")
```
Comment (DGamerL) [inline code/modules/antagonists/changeling/powers/mutations.dm:629]:
```suggestion
	if(!iscarbon(loc))
		return

	var/mob/living/carbon/C = loc
	C.throw_mode_on()

	playsound(loc, 'sound/effects/bone_break_1.ogg', 100, TRUE)
	C.visible_message("<span class='warning'>Shards of bone grow through [C.name]'s palms and fall into [C.p_their()] hands!</span>", "<span class='warning'>We expel shards of bone into our hands.</span>", "<span class='hear'>You hear organic matter ripping and tearing!</span>")
```
An early return is neater
Also, `iscarbon()` uses `istype()`, which automatically checks that the argument provided isn't `null`!

**Review by DGamerL — APPROVED** [review]

**Pull request merged**